### PR TITLE
Fix signin flow in presence of Public and Private access

### PR DESCRIPTION
**Pull request by txsmith**
I split the sign-in functionality (previously all in `signIn()`) into two functions. One (`doSignIn`) performs the redirect-flow if and only if the user clicks one of the login buttons on the home page. The other (`checkSignedIn`) checks whether the redirect-flow was done and if the user is signed in properly.

If a user was not signed in (no GitHub token in local-store and no redirect-flow ongoing), then the user is redirected to the home page again.

Please check thoroughly!

### CODE REVIEW

**Comment by eanker** [comment]
It all works as intended, and code wise I could not find something wrong, except one nit-picky comment.

Just one remark, when clicking on home we are automatically logged in, so we can not see the homepage anymore. Is this desired?

File: src/ajax/firebase-authentication.html [hunk 1]
```diff
@@ -70,51 +70,76 @@
...
         this.$.auth.auth.getRedirectResult().then(function(result) {
-          if (!result.credential) {
-            if (this.signedIn && this._tokens.GitHub) {
-              this.fire('github-token', this._tokens.GitHub);
-              this.loading = false;
-            } else if (shouldRedirect) {
-              this._redirectToGitHub(privateRepos);
-            }
-          } else {
+          redirectDone = !!result.credential
```
Comment (eanker) [inline src/ajax/firebase-authentication.html]: This is the first occurrence of redirectDone, why is it not declared?
And the ; is missing.
Just change it to: `var redirectDone = !!result.credential;`

**Comment by txsmith** [comment]
Please squash merge with the first commit message if OK

File: src/projects/feature-bar.html [hunk 2]
```diff
@@ -0,0 +1,102 @@
+<!-- Copyright (c) 2016, preview-code
```
Comment (eanker) [inline src/projects/feature-bar.html]: Was this an intended push?
I do not see this file used.

Comment (txsmith) [reply src/projects/feature-bar.html]: Nope, oops

File: src/preview-app.html [hunk 3]
```diff
@@ -190,7 +190,9 @@
     },
     attached: function() {
       // On page-load, we may or may not be signed in.
-      this.$.firebase.checkSignedIn();
+      if (this.page.page !== '') {
```
Comment (eanker) [inline src/preview-app.html]: This doesn't work, we can not get off the homepage.
So, we we click login we are redirected to the homepage and there is nothing we can click on to continue.

**Review by eanker — CHANGES_REQUESTED** [review]

File: src/preview-app.html (hunk 3 above)
Comment (txsmith) [reply src/preview-app.html]: Done

File: src/ajax/firebase-authentication.html (hunk 1 above)
Comment (txsmith) [reply src/ajax/firebase-authentication.html]: Done

**Review by eanker — APPROVED** [review]

**Comment by eanker** [comment]
Works as intended.

This pull request even fixes #366

File: src/ajax/firebase-authentication.html [hunk 4]
```diff
@@ -70,55 +70,81 @@
           type: Boolean,
           notify: true
         },
-        _tokens: Object
+        signedIn: Boolean,
+        _tokens: Object,
+        waitingForSignIn: Boolean
```
Comment (TimvdLippe) [inline src/ajax/firebase-authentication.html:75]: Maybe you can also take advantage of [`statusKnown`](https://github.com/firebase/polymerfire/blob/master/firebase-auth.html#L115-L124)

**Pull request merged**